src/erlastic_search.erl

```diff
@@ -648,6 +650,30 @@ aliases(Body) ->
 aliases(Params, Body) ->
     erls_resource:post(Params, filename:join([<<"_aliases">>]), [], [], erls_json:encode(Body), Params#erls_params.http_client_options).
 
+%% Bulk operations with default params
+-spec bulk_operation(list()) -> {ok, list} | {error, any()}.
+bulk_operation(OperationIndexTypeIdJsonTuples) ->
+    bulk_operation(#erls_params{}, OperationIndexTypeIdJsonTuples).
+
+%% Documents is [ {Index, Type, Id, Json}, {Index, Type, Id, HeaderInformation, Json}... ]
+-spec bulk_operation(#erls_params{}, list()) -> {ok, list()} | {error, any()}.
+bulk_operation(Params, OperationIndexTypeIdJsonTuples) ->
+    Body = lists:map(fun
+                       Build({delete, Index, Type, Id}) ->
+                         Build({delete, Index, Type, Id, [], []});
+                       Build({delete, Index, Type, Id, HeaderInformation}) ->
+                         Build({delete, Index, Type, Id, HeaderInformation, []});
+                       Build({Operation, Index, Type, Id, Doc}) ->
+                         Build({Operation, Index, Type, Id, [], Doc});
+                       Build({Operation, Index, Type, Id, HeaderInformation, Doc}) ->
+                         Header = build_header(Operation, Index, Type, Id, HeaderInformation),
+                         DocBin = maybe_encode_doc(Doc),
+                         Json = maybe_add_doc_for_update(DocBin),
+                         [Header, <<"\n">>, Json, <<"\n">>]
+                     end, OperationIndexTypeIdJsonTuples),
+
+    erls_resource:post(Params, <<"/_bulk">>, [], [], iolist_to_binary(Body), Params#erls_params.http_client_options).
```

Comment:
Looks good, but now the `bulk_index_docs_header/2` fun should be DRY-ed up to use this one instead :)

Reply:
You mean  bulk_index_docs_header/2 or bulk_index_docs/2 ?

Reply:
Sorry bulk_index_docs/2, bad copy-pasta :)

Reply:
okay thanks :)